XenModel/Actions/MultipleAction.cs

```diff
@@ -127,15 +143,26 @@ private void SubActionChanged(ActionBase sender)
             {
                 SubActionTitle = subAction.Title;
                 SubActionDescription = subAction.Description;
+                RecalculatePercentComplete();
                 OnChanged();
             }
         }
 
+        private void RecalculatePercentComplete()
+        {
+            int total = 0;
+            int n = subActions.Count;

```

Comment:
I believe we should be doing the calculation with (actionsByConnection.Count+actionsWithNoConnection.Count) instead of subActions.Count, because actionsByConnection does not include actions with disconnected connections, so it may happen that we'll be calculating the percentage waiting for the completion of actions that never actually started.